Helper/Order.php

```diff
@@ -881,80 +952,83 @@ public function updateOrderPayment($order, $transaction = null, $reference = nul
 
         // preset default payment / transaction values
         // before more specific changes below
-        $amount = $transaction->amount->amount;
+        $amount = @$newCapture->amount->amount ?: $transaction->amount->amount;
```

Comment:
It seems like to me that this code at times will be trying to reference data properties from null.  I believe `$newCapture` can be null.  Therefore, you have `null->amount->amount`.  While `@` will silence this error, I feel like it quietly halts the script and is a big error.  

What am I missing here?

Reply:
@mccottry 
The only case when the amount to process (used at several places after this point in the code ) is not the one stored in $transaction->amount->amount is when the transaction is of type (partial) CAPTURE. $newCapture is an object (processing capture case) or null (other cases) previously determined. So if there is no capture to process the error is silenced and $transaction->amount->amount used.
The code is too abbreviated, thus confusing, changing it to:
`
if ($newCapture) {
            $amount = $newCapture->amount->amount;
        } else {
            $amount = $transaction->amount->amount;
        }
`

Reply:
@danac-gs I've learned that `@null->amount->amount ?:` does not throw a run time error and will evaluate to false.  Interesting  Thanks!